sklearn/decomposition/incremental_pca.py

```diff
@@ -204,42 +209,44 @@ def partial_fit(self, X, y=None):
         else:
             self.n_components_ = self.n_components
 
-        if (self.components_ is not None) and (self.components_.shape[0]
-                                               != self.n_components_):
+        if (self.components_ is not None) and (self.components_.shape[0] !=
+                                               self.n_components_):
             raise ValueError("Number of input features has changed from %i "
                              "to %i between calls to partial_fit! Try "
-                             "setting n_components to a fixed value." % (
-                                 self.components_.shape[0], self.n_components_))
+                             "setting n_components to a fixed value." %
+                             (self.components_.shape[0], self.n_components_))
 
+        # This is the first partial_fit
```

Comment:
I read #3907 last week: I don't think consensus was reached. In a way, there is no contract with the final user. Last comment by @jnothman

> The point is that the interaction between fit and partial_fit hasn't been previously addressed. The assumption is that the user will choose one or the other, but not both. Forcing a particular behaviour by way of the proposed tests will make the matter more consistent, but is also perhaps more of a framework than is actually needed, because the case is somewhat pathological.

I think @ogrisel is planning to put some thought into the possibility of an API for partial fitting, which would settle the semantics for all the case.